File: 2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.js
/**
 * @param {number} n
 * @param {number[][]} roads
 * @return {number}
 */
var maximumImportance = function(n, roads) {
    let res = 0, cost = 1;
    let conn = new Array(n).fill(0);

    for (const road of roads) {
        conn[road[0]]++;
        conn[road[1]]++;
    }

    conn.sort((a, b) => a - b);

    for (const con of conn) {
        res += con * cost++;
    }
    return res;
};

File: 2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        long res = 0, cost = 1;
        vector<int> conn(n, 0);
        for (auto road : roads) {
            conn[road[0]]++;
            conn[road[1]]++;
        }
        sort(conn.begin(), conn.end());
        for (auto con : conn) res += con * (cost++);
        return res;
    }
};